src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1162,8 +1162,16 @@ private async Task HandleBreakModeEvent(ResultEventArgs results, BreakRequest br
                 string bkptno = results.Results.FindString("bkptno");
                 ulong addr = cxt.pc ?? 0;
 
+                EXCEPTION_INFO exceptionInfo;
                 bool fContinue;
                 TupleValue frame = results.Results.TryFind<TupleValue>("frame");
+
+                // test -- need to delete
+                if (ExceptionManager.TryGetExceptionBreakpoint(bkptno, out exceptionInfo) == Constants.S_OK)
+                {
+                    _callback.OnException(thread, exceptionInfo.bstrExceptionName, "", exceptionInfo.dwCode,exceptionInfo.guidType);
```

Comment:
The reason why everything is seen as an uncaught exception is because `state` was not set here.

Everything is defaulting to `ExceptionBreakpointStates.None` which leads to `EXCEPTION_STOP_SECOND_CHANCE`

```suggestion
                    _callback.OnException(thread, exceptionInfo.bstrExceptionName, "", 0, exceptionInfo.guidType, state);
```